frontend/src/components/widgets/UserAvatarLink.jsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Avatar from "@material-ui/core/Avatar";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/styles";
+import { getUrl as getUserBoardUrl } from "./LinkToUserBoard";
+
+const useStyles = makeStyles({
+  avatarLink: {
+    textDecoration: "none",
+  },
+  container: {
+    display: "flex",
+    alignItems: "center",
+    gap: 8,
+  },
+});
+
+export function getAvatarTextInitials(email) {
+  const customPartOfAnEmail = email.substr(0, email.indexOf("@"));
```

Comment:
customPartOfAnEmail ?? I shook my head around for some time, but what I see is you getting the prefix of an email

If my email is vuyisanani.meteni@umuzi.org,  you only getting/storing vuyisanani.meteni in your variable, not entirely sure if your variable more descriptive for that

Reply:
perhaps use `emailUsername`

Reply:
emailUsername is perfect